api/capi/src/nntrainer.cpp

```diff
@@ -888,7 +888,41 @@ ml_train_dataset_set_property_for_mode_(ml_train_dataset_h dataset,
         return status_;
       }
 
-      status_ = db->setProperty(args);
+      std::vector<std::string> properties;
+      for (unsigned int i = 0; i < args.size(); ++i) {
+        char *key_ptr = (char *)args[i];
+        std::string key = key_ptr;
+        std::string value;
+
+        /** Handle the user_data as a special case, serialize the address and
+         * pass it to the databuffer */
+        if (key == "user_data") {
```

Comment:
PropsUserData::key?

Reply:
also istrequal()?

Reply:
update to handle lower case and whitespace case, for the below comment :)
To decouple headers, I'll keep "user_data" here.

Reply:
capi is allowed to depend on nntrainer sources, so moving the key to a header and using here should be fine.
otherwise create `user_data` as a static const variable here and for line 915.

Reply:
Sure, I'll change to static const.